activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -275,6 +277,14 @@ def convert_value(value, options = {})
           value
         end
       end
+
+      def set_defaults(target, source = self)
```

Comment:
I originally used this for the `new_from_hash_copying_default` as well, but realized later I didn't need it. I'll remove it.